Producing a blog on Business Operations and Software Testing. (#293)
ScottLogic/blog: _posts/2025-01-25-how-business-operations-and-software-testing-overlap.md

---
title: How Business Operations and Software Testing Overlap
date: 2025-01-16 00:00:00 Z
categories:
  - Testing
tags:
  - Testing
summary: This blog explores the overlap between life in Business Operations and Software Testing.  It examines the distinct skillsets required for each discipline and how they are applied in the world of business.
author: lperrett
---

## Overview

I have recently completed a secondment to the Business Operations (Bus Ops) department within Scott Logic. Before I joined the team, I did not have any understanding of what the team did. However, that understanding soon changed within a month of joining and continued to uncover new paths for the duration of my secondment.

> When I joined the team, I was astounded as to how much reporting is required within the Bus Ops team. I quickly decided to buy a course on data analysis with the view of understanding how data analysts properly analyse data. There was an added determination that I would like to take this to the next stage and start studying data science, which I believe is the next big thing as it slides directly into Artificial Intelligence (AI) and Machine Learning (ML). I really believe we should now start to focus our efforts on AI and ML, as they are the next big players within the industry.

Reviewer: ```suggestion
When I joined the team, I was astounded as to how much reporting is required within the Bus Ops team. I quickly decided to buy a course on data analysis with the intention of understanding how data analysts properly analyse data. There was an added determination that I would like to take this to the next stage and start studying data science, which I believe is the next big thing as it slides directly into Artificial Intelligence (AI) and Machine Learning (ML). I really believe we should now start to focus our efforts on AI and ML, as they are the next big players within the industry.
```
Fits slightly better.

## Discussion: Business Operations

I found out that the Bus Ops department is classed as "silent running". This means that they always help to protect the business, make sure reports are sent out on time, legalities are taken care of, staff have work to do and that they are contactable and approachable at any point in time. The team needs to be silent running because the business (our own company) is their client, and they need to ensure the business runs smoothly. Bus Ops want their consultants and clients to have a good experience at all times. Therefore, when Bus Ops activities run smoothly in the background it helps the business to run smoothly in the foreground. This shows they play a vital role in the organisation and perhaps go unnoticed.

> On my chosen course I learned definitions such as: "Big Data, Data Cleansing, Data Marts, Data Lakes, Data Pipelines" as well as learning what it is like to be a data analyst. It really helped me to understand what I could expect in the department. These are all terms I had heard before, but I never had to dive deeper until now.

Reviewer: ```suggestion
On my chosen course I learned definitions such as "Big Data", "Data Cleansing", "Data Marts", "Data Lakes", and "Data Pipelines", as well as learning what it is like to be a data analyst. It really helped me to understand what I could expect in the department. These are all terms I had heard before, but I never had to dive deeper until now.
```
I think if you want quotes around this section it should be for each term rather than the group.

When I didn't complete certain activities on time, I could see the impact on the business. Stakeholders did not get their messages on time, so financial reporting and tracking periods were impacted. There was a sense of responsibility that the data meant something to others, even if I did not fully understand it at the time. However, it soon created a story of "Why does this task need to be done?", which helped promote efficiency in delivery.

> I continued my expansion into the team and learned about ISO 27001. ISO 27001 is an international standard to manage information security. I became responsible for auditing tickets relating to Change Management, Risk Management and ISO Standards. There was a precedence set that every month and every quarter these meetings would take place, and suddenly I was drifted back into the world of auditing, to what I had done in former years with the Capability Maturity Model Integration (CMMI) and Test Maturity Model Integration (TMMI) in previous testing roles. The emphasis being to maintain standards that had been implemented and report any deviations to the team and external auditor.

Reviewer: ```suggestion
I continued my expansion into the team and learned about ISO 27001. ISO 27001 is an international standard to manage information security. I became responsible for auditing tickets relating to Change Management, Risk Management and ISO Standards. There was a precedence set that every month and every quarter these meetings would take place, and suddenly I was drifted back into the world of auditing, to what I had done in former years with the Capability Maturity Model Integration (CMMI) and Test Maturity Model Integration (TMMI) in previous testing roles. The emphasis was on maintaining standards that had been implemented and reporting any deviations to the team and external auditor.
```
Tense changes

### Documentation

Documentation is crucial in software testing as it allows planning, designing, execution, tracking and closure to take place. It can include errors, so it is important to keep any documents up to date, where possible. Documentation can be used more loosely to provide guidance, such as coding standards, process diagrams and ways of working.

In the planning phase, Test Strategies and Test Plans will be created to outline the overall approach for the project. Test Design can include Testware documentation, but it can also be about thinking critically, challenging assumptions and being creative to uncover information. Execution can occur with test ideas, but they don’t have to be documented. If Testware exists, then tests need to be recorded as passed/failed and have associated defect reports. During the closure phase, a Test Summary report documents a summary of the testing activities and an outcome of the overall testing process.

### Problem Solving

> Problem Solving is a critical part of software testing, it helps testers to address challenges and ensure good quality software is being produced. It helps testers work out whether something is a defect, handle ambiguous requirements by learning to ask the "right" questions, find the root cause as to why something occurred, create better Testware, deal with environment issues, write automated scripts, manage test data and collaborate with others to explain the testing stance.

Reviewer: ```suggestion
Problem Solving is a critical part of software testing, it helps testers to address challenges and ensure good quality software is being produced. It helps testers work out whether something is a defect, handles ambiguous requirements by learning to ask the "right" questions, and finds the root cause as to why something occurred. They can create better Testware, deal with environment issues, write automated scripts, manage test data and collaborate with others to explain the testing stance.
```
Thought this sentence was a bit long, and it switches whether the words feel natural as singular or plural.

Reviewer 2: I think the original is more correct; the phrase here is "testers handle ambiguous requirements", not "Problem Solving handles ambiguous requirements"

Author: Keeping the original.

Acceptance of ideas and concepts can be more challenging as it leads to a subjective nature. Teams often face different viewpoints typically based on role or their worldview. Ultimately, this means that understanding the situation is vital to ensure that the best decisions are being made based on the knowledge they have at the time of the decision. Of course, decisions can be made, points can be proved, and changes can occur later on. Sometimes compromises must be made, alternatively majority decisions need to be taken forward, which can be cognitively or emotionally difficult if someone fundamentally disagrees or can see how the decision will impact the project further down the line. In addition, trade-offs may occur due to a range of issues, e.g. time and budget being the most common, and testers can acknowledge that business decisions, priorities or constraints take precedence, and testers can adopt their strategies when needed.

### Making mistakes

> Making mistakes is critical to our growth and development. We cannot just learn from other peoples' experiences; we must learn from our own experiences as well.

Reviewer: ```suggestion
Making mistakes is critical to our growth and development as professionals. We cannot just learn from other peoples' experiences; we must learn from our own experiences as well.
```
Minor addition.
